client/src/pages/ShoppingCartPage.js

```diff
@@ -31,6 +35,46 @@ function ShoppingCartPage(props) {
 
   const [cart, setCart] = useState(location.state.shoppingCart);
 
+  const [client, setClient] = useState({});
+
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    let clientID;
+    async function updateClient() {
+      if (authContext.currentUser && authContext.currentUser.role === UserRoles.CLIENT)
+        clientID = authContext.currentUser.id;
+      else if (location.state.clientID) clientID = location.state.clientID;
+      try {
+        const fromServer = await getClientByID(clientID);
+        setClient(fromServer);
+      } catch (err) {
+        console.error(
```

Comment:
There should be an ErrorToast as in the ClientDetailsPage, remove this log and use it. 

Thank you very much for moving this logic outside the components, you were not supposed to do it, in the work context I would have rejected this change, but in this context I'm very grateful.